cpp/src/sort/segmented_sort_impl.cuh

```diff
@@ -68,12 +68,13 @@ struct column_fast_sort_fn {
                  column_view const& segment_offsets,
                  mutable_column_view& indices,
                  bool ascending,
-                 rmm::cuda_stream_view stream)
+                 rmm::cuda_stream_view stream,
+                 rmm::mr::device_memory_resource* mr)
   {
     // CUB's segmented sort functions cannot accept iterators.
     // We create a temporary column here for it to use.
     auto temp_col =
-      cudf::detail::allocate_like(input, input.size(), mask_allocation_policy::NEVER, stream);
+      cudf::detail::allocate_like(input, input.size(), mask_allocation_policy::NEVER, stream, mr);
```

Comment:
This temporary should not be created with the passed in `mr`
https://github.com/rapidsai/cudf/blob/branch-23.04/cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md#temporary-memory